src/clients/cassandra/CassandraConnector.js

```diff
@@ -93,6 +93,33 @@ function executeQuery(query, params, options) {
 }
 
 /**
+ * @param {string} query
+ * @param {string[]} params
+ * @param {int} fetchSize
+ * @param {string} pageState
+ * @returns {Promise.<object[]>}
+ */
+function executeQueryWithPageState(query, params, pageState, fetchSize) {
+  return new Promise((resolve, reject) => {
+    const DEFAULT_FETCH = 15;
+
+    let options = {
+      fetchSize: fetchSize > 0 ? fetchSize : DEFAULT_FETCH
+    };
+
+    if(pageState) {
+      options.pageState = pageState;
+    }
+
+    console.log(query);
+
+    client.execute(query, params, options)
+    .then(result => resolve(Object.assign({}, {pageState: result.pageState, rows: result.rows})))
```

Comment:
Why the need for Object.assign instead of just declaring a new object? Isn't the following simpler?

```js
.then(result => {
  const pageState = result.pageState;
  const rows = result.rows;

  resolve({
    pageState,
    rows
  })
})
```

If you want it more concise, it's also possible to inline the variables:

```js
.then(result => ({pageState: result.pageState, rows: result.rows}))
```